app/policies/application_policy.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def index?
```

Comment:
I think you can simplify these policy classes a lot - in most cases, the rules are the same for all methods. I don't want to block the PR. Try to express the rules more briefly, please. In that way, you will help someone (and especially yourself) in the future with understanding this part of code.

Reply:
I have refactored policies, thanks.